azavea/tilegarden
# Feature/cartocss integration

## Overview

Overhauls the way map configurations are specified. Database connection and map layer definitions are defined in map-config.mml, which is then converted into an xml file that mapnik can read at transpile time. Secrets are templated in from the environment in order to keep them off of version control and separate between development/production.


### Notes

Templating in environment variables is good for keeping configurations separate between development and production environments, but at the end of the day secrets still end up in an XML file on AWS. This could be changed to happen at runtime avoid having credentials saved to a file but at the cost of performance, or the XML could be encrypted with something like `bcrypt` and then decrypted at runtime.


## Testing Instructions

 * Run the development server with `./scripts/update` and `./scripts/server`. 
 * Open `src/demo/pwd_inlets_index.html` to display an example map.
 * Open `src/tiler/src/res/style.mss` and change the `marker-fill` parameter. Wait for the server to restart and refresh the demo page to see the marker color update in your browser.


Resolves #25, resolves #28
